README.md

```diff
@@ -1,55 +1,166 @@
+
 # SwitchbOTA
-Replaces the default firmware on the SwitchBot Plug Mini via OTA, enabling the use of Tasmota without disassembling the unit.
+![51HjLj15OZL _SL1500_](https://github.com/kubedzero/switchbota/assets/51148401/0bdf7ae9-9eaf-4698-819e-78625adce8fc)
+
+This project helps to replace the default firmware on the SwitchBot Smart Plug Mini via an OTA update, enabling the use of Tasmota open-source firmware without disassembling the device.
+
+## SwitchBot Hardware Details
+
+
 
-Similar to [Espressif2Arduino](https://github.com/khcnz/Espressif2Arduino).
+There are two SKUs of the SwitchBot Smart Plug Mini released in the US: 
 
-## Compatible Versions
-SwitchbOTA has been tested to work on the two current existing US plug revisions:
-- W1901400 (Original US Plug Mini, March 2022)
-- [W1901401](https://github.com/kendallgoto/switchbota/issues/19) (HomeKit US Plug Mini, June 2022)
+- The original "Smart Plug Mini" with model number W1901400, released in early 2022. As of 2023-12, Amazon sells these in [four-packs](https://www.amazon.com/dp/B09YV2L3MN) for about $30, [two-packs](https://www.amazon.com/dp/B09YV3LH8Y) for $19, or [one-packs](https://www.amazon.com/dp/B09QFLJH8T) for $12
+- The updated "HomeKit Smart Plug Mini" with model number W1901401, released later in 2022. As of 2023-12, Amazon sells these in [four-packs](https://www.amazon.com/dp/B0B9RHTM6Y) for about $38 or [one-packs](https://www.amazon.com/dp/B0B39DJFR8) for $15
 
-Running SwitchbOTA has been tested on v1.4 and lower. You may want to downgrade to V1.4 or lower before starting the process.
+Some details about the devices are as follows:
+
+- Both devices use identical circuitry. The only difference is the firmware used. More detail can be found in https://github.com/kendallgoto/switchbota/issues/19
+- They use Type B NEMA 5-15 plugs and sockets
+- They are ultrasonically welded shut, requiring destruction of the plastic housing to access the circuits within. There are no screws to allow access to the internals, the way there is with the Sonoff S31
+- They use the Shanghai Belling BL0937 energy measurement IC to track power usage
+- They use the Espressif ESP32-C3 chip for WiFi, Bluetooth Low Energy (BLE), and circuit control. See [here](https://tasmota.github.io/docs/ESP32/#esp32-differences) to learn about the different types of ESP32
+- The SwitchbOTA flashing process is confirmed to work on both device models using firmware v1.4 or older (v1.3, v1.2)
+- It appears that a plug's MAC addresses for WiFi and BLE are identical other than the last character, which is offset by two. If the BLE-MAC is `34:85:18:0F:CC:DE`, then the WIFI-MAC is likely `34:85:18:0F:CC:DC` (DC in hex is 220, DE is 222)
```

Comment:
this is standard across esp32s fwiw; https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-reference/system/misc_system_api.html#mac-address